sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,126 @@
+
+        /// <summary>
+        /// Gets the <see cref="SparkBatchJob"/>.
+        /// You should await <see cref="WaitForCompletionAsync(CancellationToken)"/> before attempting to use session in this pending state.
+        /// </summary>
+        /// <remarks>
+        /// Azure Synapse will return a <see cref="SparkBatchJob"/> immediately but may take time to the session to be ready.
+        /// </remarks>
+        public override SparkBatchJob Value => _value;
+
+        /// <inheritdoc/>
+        public override bool HasCompleted => _completed;
+
+        /// <inheritdoc/>
+        public override bool HasValue => true;
+
+        /// <inheritdoc/>
+        public override Response GetRawResponse() => _response.GetRawResponse();
+
+        /// <inheritdoc/>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<SparkBatchJob>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<SparkBatchJob>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (!_completed)
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(SparkSessionOperation)}.{nameof(UpdateStatus)}");
+                scope.Start();
+
+                try
+                {
+                    if (async)
+                    {
+                        _response = await _client.RestClient.GetSparkBatchJobAsync(_value.Id, true, cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        _response = _client.RestClient.GetSparkBatchJob(_value.Id, true, cancellationToken);
+                    }
+                    _completed = !IsJobRunning(_response.Value.Result.ToString(), _response.Value.State);
+                }
+                catch (Exception e)
+                {
+                    scope.Failed(e);
+                    throw;
+                }
+            }
+
+            return GetRawResponse();
+        }
+
+        private static bool IsJobRunning(string jobState, string livyState)
+        {
+            if (StringComparer.OrdinalIgnoreCase.Equals ("Succeeded", jobState) ||
```

Comment:
Could we take the same lookup approach here as with `livyState`?

Reply:
Possibly, I'm unsure if the capitalization matches (or is guaranteed). Asking service team.

Reply:
Service team confirms always lower case, will fix.